react-native/react/reducers/tracker.js

```diff
@@ -0,0 +1,206 @@
+      }
+      const username = action.payload.username
+
+      return {
+        ...state,
+        username
+      }
+
+    case Constants.updateProofState:
+      const proofs = state.proofs
+      const allOk: boolean = proofs.reduce((acc, p) => acc && p.state === normal, true)
+      const anyWarnings: boolean = proofs.reduce((acc, p) => acc || p.state === warning, true)
+      const anyError: boolean = proofs.reduce((acc, p) => acc || p.state === error, false)
+      const anyPending: boolean = proofs.reduce((acc, p) => acc || p.state === pending, false)
+
+      let proofState: SimpleProofState = 'error'
+
+      if (allOk) {
+        proofState = 'normal'
+      } else if (anyWarnings) {
+        proofState = 'warning'
+      } else if (anyError) {
+        proofState = 'error'
+      } else if (anyPending) {
+        proofState = 'pending'
+      }
+
+      return {
+        ...state,
+        proofState
+      }
+
+    case Constants.setProofs: 
+      if (!action.payload) {
+        return state
+      }
+
+      const identity: Identity = action.payload.identity
+
+      return {
+        ...state,
+        proofs: identity.proofs.map(rp => remoteProofToProof(rp.proof))
+      }
+
+    case Constants.updateProof:
+      if (!action.payload) {
+        return state
+      }
+
+      const rp: RemoteProof = action.payload.remoteProof
+      const lcr: LinkCheckResult = action.payload.linkCheckResult
+      return {
+        ...state,
+        proofs: updateProof(state.proofs, rp, lcr)
+      }
+
+    case Constants.updateUserInfo:
+      if (!action.payload) {
+        return state
+      }
+      return {
+        ...state,
+        userInfo: action.payload
+      }
+
+    case Constants.registerIdentifyUi:
+      return {
+        ...state,
+        serverStarted: action.payload && !!action.payload.started || false
+      }
+
+    case Constants.markActiveIdentifyUi:
+      return {
+        ...state,
+        serverActive: action.payload && !!action.payload.active || false
+      }
+
+    default:
+      return state
+  }
+}
+
+function mapTagToName (obj: any, tag: any): ?string {
+  return Object.keys(obj).filter(x => obj[x] === tag)[0]
+}
+
+function stateToColor (state: SimpleProofState): string {
+  if (state === normal) {
+    return 'green'
+  } else if (state === warning) {
+    return 'yellow'
+  } else if (state === error) {
+    return 'red'
+  }
+
+  return 'gray'
+}
+
+function proofStateToSimpleProofState (proofState: ProofState): SimpleProofState {
+  const statusName: ?string = mapTagToName(identify.ProofState, proofState)
+  switch (statusName) {
+    case 'ok':
+      return normal
+    case 'tempFailure':
+    case 'superseded':
+    case 'posted':
+      return warning
+    case 'revoked':
+    case 'permFailure':
+      return error
+    case 'looking':
+    case 'none':
+    default:
+      return pending
+  }
+}
+
+function remoteProofToProof (rp: RemoteProof, lcr: ?LinkCheckResult): Proof {
```

Comment:
@chrisnojima I decided to put the type transform functions the reducer because:

1. actions already have enought code in them as is, and they deal with all the side effecty stuff.

2. this fits well with the reducers doing data transformations, and all of these are pure funcitons.